src/apps/peer/udpserver.c

```diff
@@ -112,8 +112,12 @@ static int udp_create_server_socket(server_type *server, const char *ifname, con
   return 0;
 }
 
-static server_type *init_server(int verbose, const char *ifname, char **local_addresses, size_t las, int port) {
+static server_type *init_server(int verbose, const char *ifname, char **local_addresses, size_t las, uint16_t port) {
 
+  if(port == 65535){
+    return NULL;
+  }
+  
   server_type *server = (server_type *)malloc(sizeof(server_type));
 
   if (!server) {
```

Comment:
instead of `return server` here, it should probably be `return NULL` to more clearly indicate we're returning due to allocation failure.